agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -61,27 +67,66 @@ public static Configuration create(Path agentJarPath) throws IOException {
         Configuration config = loadConfigurationFile(agentJarPath);
         PreviewConfiguration preview = config.instrumentationSettings.preview;
 
-        preview.roleName = overlayWithEnvVar(APPLICATIONINSIGHTS_ROLE_NAME, WEBSITE_SITE_NAME, preview.roleName);
+        preview.roleName = overlayWithEnvVars(APPLICATIONINSIGHTS_ROLE_NAME, WEBSITE_SITE_NAME, preview.roleName);
         preview.roleInstance =
-                overlayWithEnvVar(APPLICATIONINSIGHTS_ROLE_INSTANCE, WEBSITE_INSTANCE_ID, preview.roleInstance);
+                overlayWithEnvVars(APPLICATIONINSIGHTS_ROLE_INSTANCE, WEBSITE_INSTANCE_ID, preview.roleInstance);
 
+        loadSamplingRateEnvVar(preview);
+        loadLogCaptureEnvVar(preview);
+        loadJmxMetrics(preview);
+
+        return config;
+    }
+
+    private static void loadSamplingRateEnvVar(PreviewConfiguration preview) {
+        final String samplingRate = overlayWithEnvVar(APPLICATIONINSIGHTS_SAMPLING_RATE, "100");
+        if (!samplingRate.equals("100")) {
+            FixedRateSampling fixedRateSampling = new FixedRateSampling();
+            fixedRateSampling.percentage = Double.parseDouble(samplingRate);
+            Sampling sampling = new Sampling();
+            sampling.fixedRate = fixedRateSampling;
+            preview.sampling = sampling;
+        }
+    }
+
+    private static void loadLogCaptureEnvVar(PreviewConfiguration preview) {
+        final String logCaptureThreshold = overlayWithEnvVar(APPLICATIONINSIGHTS_LOG_CAPTURE_THRESHOLD, "INFO");
+        if (!logCaptureThreshold.equals("INFO")) {
```

Comment:
same here, i don't think it will handle correctly json value "WARN" and env var "INFO"